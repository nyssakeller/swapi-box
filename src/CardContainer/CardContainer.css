.Cards {
  display: flex;
  flex-direction: row;
  vertical-align: top;
}