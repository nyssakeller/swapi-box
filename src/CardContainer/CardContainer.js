import React from 'react';
import Card from '../Card/Card.js';
import './CardContainer.css';


const CardContainer = ({data, favoriteCard}) => {
  let renderedCards = data.map((set, key) => 
    <Card name={set.name}
          description={set.description}
          type={set.type}
          numer={set.number}
          favoriteCard={favoriteCard}
          id={set.name}
          key={key} />
  )

  return(
    <section className='cards'>
      {renderedCards}
    </section>
  )
}

export default CardContainer;