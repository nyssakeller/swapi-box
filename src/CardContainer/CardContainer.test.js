import React from 'react';
import { shallow } from 'enzyme';
import CardContainer from './CardContainer.js';

const mockData = [
  {name: 'vehicle1', type: 'toyota', description: 'rav4', number: 5},
  {name: 'vehicle1', type: 'chevy', description: 'equinox', number: 6}
]

describe('CardContainer', () => {

  it('match snapshot', () => {
    const wrapper = shallow(<CardContainer 
                              data={mockData} />)
    expect(wrapper).toMatchSnapshot();
  })

  it('card for every object in the dataArray', () => {
    const wrapper = shallow(<CardContainer 
                              data={mockData} />)
    expect(wrapper.find('Card').length).toEqual(2);
  })

})