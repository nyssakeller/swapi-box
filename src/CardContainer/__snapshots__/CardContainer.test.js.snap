// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    data={
        Array [
            Object {
              "description": "rav4",
              "name": "vehicle1",
              "number": 5,
              "type": "toyota",
            },
            Object {
              "description": "equinox",
              "name": "vehicle1",
              "number": 6,
              "type": "chevy",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          description="rav4"
          favoriteCard={undefined}
          id="vehicle1"
          name="vehicle1"
          numer={5}
          type="toyota"
/>,
        <Card
          description="equinox"
          favoriteCard={undefined}
          id="vehicle1"
          name="vehicle1"
          numer={6}
          type="chevy"
/>,
      ],
      "className": "cards",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "description": "rav4",
          "favoriteCard": undefined,
          "id": "vehicle1",
          "name": "vehicle1",
          "numer": 5,
          "type": "toyota",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "description": "equinox",
          "favoriteCard": undefined,
          "id": "vehicle1",
          "name": "vehicle1",
          "numer": 6,
          "type": "chevy",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            description="rav4"
            favoriteCard={undefined}
            id="vehicle1"
            name="vehicle1"
            numer={5}
            type="toyota"
/>,
          <Card
            description="equinox"
            favoriteCard={undefined}
            id="vehicle1"
            name="vehicle1"
            numer={6}
            type="chevy"
/>,
        ],
        "className": "cards",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "description": "rav4",
            "favoriteCard": undefined,
            "id": "vehicle1",
            "name": "vehicle1",
            "numer": 5,
            "type": "toyota",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "description": "equinox",
            "favoriteCard": undefined,
            "id": "vehicle1",
            "name": "vehicle1",
            "numer": 6,
            "type": "chevy",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
